### ISSUE TITLE
Elgato Stream Deck Pedal

### ISSUE THREAD

**Issue by berryv97** [bitfocus/companion#1902]
### Is this a feature relevant to companion itself, and not a module?

- [X] I believe this to be a feature for companion, not a module

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the feature

Elgato has just released a new product the "Stream Deck Pedal", following the logic of companion, it would be nice to have integrated power on companion.

Secondly, controlling the depression of the pedal in companion would be a good idea.

https://www.elgato.com/fr/stream-deck-pedal

### Usecases

Being able to activate macros with the feet.

**Comment by Julusian** [comment]
This is planned, and is waiting on one of the core devs receiving a pedal or for help with support at https://github.com/Julusian/node-elgato-stream-deck/issues/36

**Comment by martinszugat** [comment]
Hi Julusian,

I've tested v2.2 but Companion didn't find the Pedal device. So I assume it is not supported, yet. Right?

Best,

Martin

**Comment by josephdadams** [comment]
Correct, not yet supported @martinszugat

**Comment by Julusian** [comment]
@martinszugat when it is supported this issue will be closed. you will get an email from github when that happens

**Comment by velocity23** [comment]
+1 from me - this would be super useful

**Comment by Julusian** [comment]
In theory the pedal is supported in the betas, but I haven't had confirmation from anyone if it does or does not work

**Comment by ehfyn** [comment]
Super useful feature to have the pedal support! We would use it in our production to control for example PowerPoint, teleprompters etc.
It would be gold if an implementation would work with the Satellite feature on Raspberry Pi’s, since we have deployed a handful of those in our studios and in our control rooms.

Any idea when it will be included in a stable release?

**Comment by Julusian** [comment]
It is already included in the latest build of satellite https://github.com/bitfocus/companion-satellite/releases/tag/v0.5.0-0, but I havent heard from anyone whether it works.
Assuming the changes in beta work, then it will be in the 2.2.1 release that will hopefully be done within a couple of weeks

**Comment by ehfyn** [comment]
I have just ordered a pedal - will test it, hopefully soon!

**Comment by velocity23** [comment]
I’ve got a pedal as well, if there’s anything specifically the team needs tested feel free to let me know!

**Comment by liamsomerville** [comment]
@velocity23 my use case would be with a atem video switcher to change the program source

Is there even a remote possibility you have an atem to test with?

**Comment by Julusian** [comment]
Someone on the facebook group has tried a pedal and reports that companion doesnt detect it. I believe @JeffreyDavidsz has ordered one, so hopefully I can get some help from them in figuring out what is wrong.

@liamsomerville it doesnt need to be tested with an atem specifically, once it works in companion at all it will work just like any other stream deck

**Comment by kraves-oz** [comment]
@velocity23 can you download the latest beta and see if it works.

**Comment by velocity23** [comment]
Not being recognized as a USB device on the latest beta

**Comment by JeffreyDavidsz** [comment]
Missed the mailman today, pedal shall arrive tomorrow.could be as simple as a different product ID. I'll check it

**Comment by ehfyn** [comment]
Also tested here, but only from Companion Satellite (RPI4) - nothing shows in the "Surfaces" list.
How should it be detected, and how is it planned to configure it within Companion? (i don't imagine an image of a pedal, like on the elgato software?)

**Comment by Julusian** [comment]
Once it works, it will show up as a surface, and behave as if it is a 3x1 button streamdeck. Nothing else special will be shown.

An important feature to be aware of, is that in the settings for the pedal in the surfaces tab of companion, is that you are able to adjust the position within the grid, so that it isnt using buttons from the top left corner. 
For example, you could move it to the top right, so that it can use the space which is unused on a 15 key streamdeck
![image](https://user-images.githubusercontent.com/1327476/167407535-5c43bb3b-c8eb-49d3-b4d3-76d3284e890c.png)

**Comment by ehfyn** [comment]
Ahh okay, that makes perfectly sense! :) Hope to have it working with satellite soon, since it's a killer feature! I'm currently looking into modding the pedal, so it becomes a bit more silent in operation, so our mics don't pick it up so easy.

**Comment by JeffreyDavidsz** [comment]
Hi guys, update; we have updated the beta with a fix. My pedal is working now to. Small reminder; if you have a page up programmed on 1.1 (which is default) the left pedal will click you to the second page ;-) So I use the x offset hahaha
